### PR TITLE
[4186] Admin funding view

### PR DESCRIPTION
**Pull request by AbigailMcP**
### Context

https://trello.com/c/9euPhMgK/4186-admin-funding-view

### Changes proposed in this pull request

- Adds two new controllers `SystemAdmin::Funding::PaymentSchedulesController` and `SystemAdmin::Funding::LeadSchoolsController` that serve data for four new pages on following URLs:

  * /providers/id/funding/payment-schedule
  * /lead-schools/id/funding/payment-schedule
  * /providers/id/funding/trainee-summary
  * /lead-schools/id/funding/trainee-summary

- Reuses the view objects and existing functionality for the views
- Creates new view object `Funding::NavigationView` to deal with the logic for rending the correct tab links, depending on whether you're an admin / viewing a lead school / viewing a provider
- Adds a links to funding from provider and lead school show pages
- Adds a new table to providers show page to show some information about the provider you're looking at

### Guidance to review

- Sign in a system admin
- Go to providers -> select a provider
- Click 'View funding' link
- See that the correct payment schedule is displayed
- Click 'Trainee summary'
- See that the correct trainee summary is displayed

Repeat the above steps for a lead school (choose one that's linked to another persona)

Screenshot 2022-06-06 at 17 15 59

Screenshot 2022-06-06 at 17 16 09

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by HettieS** [comment]
Not a massive deal - but I wonder if we should hide 'Funding' in the nav when the current user is system admin?

Screenshot 2022-06-07 at 12 01 29

**Review by HettieS — APPROVED** [review]
Looks good, just a typo in a file name and maybe hiding 'Funding' in the nav if possible for admins!

File: app/view_objects/funding/navigation_view.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      return funding_trainee_summary_path if !system_admin
+
+      if organisation.is_a?(Provider)
+        provider_funding_trainee_summary_path(organisation)
+      else
+        lead_school_funding_trainee_summary_path(organisation)
+      end
+    end
+
+  private
```
[the remark this answers is not in the document]
Comment (gpeng) [reply app/view_objects/funding/navigation_view.rb:36]: This will make the attr_reader below private.

**Comment by gpeng** [comment]
Not a massive issue but the back links from the payment schedule and summary pages go back to the home page rather than the provider/school page.

File: app/view_objects/funding/navigation_view.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Funding
+  class NavigationView
+    include Rails.application.routes.url_helpers
```
[the remark this answers is not in the document]
Comment (AbigailMcP) [reply app/view_objects/funding/navigation_view.rb:5]: Discussed - this is to give us access to the Rails path helpers.

**Comment by AbigailMcP** [comment]
> Not a massive issue but the back links from the payment schedule and summary pages go back to the home page rather than the provider/school page. We could card that up as a bug and get this in if it's fiddly.

Hettie and I chatted about this and we're gonna card it up - it's not difficult but we want to check what should happen with the back links for normal users too (currently it just goes back to root_path from where ever they were)

**Pull request merged**